fabric-network/lib/contract.js

```diff
@@ -197,6 +198,33 @@ class Contract {
 		return listener;
 	}
 
+	/**
+	 * Provide the Discovery Interest settings to help the peer's discovery service
+	 * build an endorsement plan. This interest must include this contract's
+	 * chaincode name and any chaincode names that this chaincode may call.
```

Comment:
The contract already knows its chaincode name. Does this really need to be passed in by the user or could we include it?

Reply:
If we were to include it for the user, then we have to add another process to add collections for 'this' chaincodeId, thus complicating how the user applies their interest. I will post this question in the JIRA.

Reply:
How about if we search through the list and if we do not find it, we add it. Then it would still be just one call to add and still allow them to include this contract's chaincode name if they wish to associate the collections for it.